src/editor/StyleEditor.cpp

```diff
@@ -326,4 +353,117 @@ QStringList getStyles() {
   return styles;
 }
 
+#if defined(Q_OS_WINDOWS)
+void customizeNativeTitleBar(WId w) noexcept {
+  if (QOperatingSystemVersion::current() >=
+      QOperatingSystemVersion::Windows10) {
+    static ULONG osBuildNumber = *reinterpret_cast<DWORD *>(0x7FFE0000 + 0x260);
+    // https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/ntddk/ns-ntddk-kuser_shared_data#syntax
+    // member "NtMinorVersion" of _KUSER_SHARED_DATA. don't even ask where the
+    // address comes from
```

Comment:
could you explain a bit how you got the address

Reply:
it's where the KUSER_SHARED_DATA struct resides in top of the first chunk of process space that the OS reserves for its usage. it's kind of a hack to avoid using GetVersionEx/RtlGetVersion which both require a proper app manifest which is hard to get using cmake if you want to keep it working on all generators... trust me, i tried

Reply:
tl;dr it's a constant & it comes from the nt kernel api